maven-plugin/src/main/java/org/openmrs/maven/plugins/utility/DefaultWizard.java

```diff
@@ -711,26 +713,88 @@ private String getDefaultDbUri(Server server, DockerHelper dockerHelper) {
         return dbUri;
     }
 
-    private void promptForDockerHostIfMissing(DockerHelper dockerHelper) {
+    private void promptForDockerHostIfMissing(DockerHelper dockerHelper, String dockerHost) {
         if (StringUtils.isBlank(dockerHelper.getDockerHost())) {
-            showMessage("Docker Host URL is needed in order to connect to your Docker instance.");
-            String dockerHost;
+            String host = null;
             if(SystemUtils.IS_OS_LINUX){
-                dockerHost = promptForValueIfMissingWithDefault("Please specify Docker host URL (either 'tcp://' or 'unix://')",
-                        dockerHelper.getDockerHost(), "dockerHost", DockerHelper.DEFAULT_HOST_LINUX);
-            } else {
-                showMessage("Please note that neither 'Docker for Mac' nor 'Docker for Windows' are currently supported.");
+                host = promptForValueIfMissingWithDefault("Please specify Docker host URL (either 'tcp://' or 'unix://')",
+                        dockerHost, "dockerHost", "");
+            }
+            else if (SystemUtils.IS_OS_WINDOWS){
+                showMessage("Checking default Windows Docker host addresses:");
 
-                String defaultDockerHost = getDefaultDockerHost();
+                showMessage("Checking \"Docker for Windows\"");
+                if (isDefaultDockerForWindowsPresent()) {
+                    host = DockerHelper.DEFAULT_HOST_DOCKER_FOR_WINDOWS;
+                }
+                else {
+                    showMessage("Checking \"Docker Toolbox\"");
+                    String dockerToolboxHost = getDefaultDockerToolboxHost();
+                    if (dockerToolboxHost != null) {
+                        host = dockerToolboxHost;
+                    }
+                }
 
-                dockerHost = promptForValueIfMissingWithDefault(
-                        "Please specify Docker host URL", dockerHelper.getDockerHost(), "dockerHost", defaultDockerHost);
+                if (host == null) {
+                    showMessage("Docker Host URL is needed in order to connect to your Docker instance.");
+                    host = promptForValueIfMissingWithDefault("Please specify Docker host URL",
+                            host, "dockerHost", "");
+                }
+            }
+            else if (SystemUtils.IS_OS_MAC_OSX) {
+                showMessage("Checking default Docker for Mac host address:");
+                if (isDefaultDockerForMacPresent()) {
+                    host = DockerHelper.DEFAULT_HOST_DOCKER_FOR_MAC;
+                }
```

Comment:
@rkorytkowski 

Thanks for code review :)
I've refactored code responsible for `-DdockerHost` property.
Now it behaves like that:

**1. IF Running for first time WITHOUT -DdockerHost specified**
- Try to find docker host by checking for default addresses and assign it to `sdk.properties`
- If docker address is invalid throw exception with message You suggested above

**2. IF Running NOT for first time WITHOUT -DdockerHost specified**
- Use docker host specified as property in sdk.properties
-  If `dockerHost` in `sdk.properties` is not found, behave same as in point **1.**

**3. IF Running WITH -DdockerHost specified WITHOUT value**
- Behave same as in point **1.**

**4. IF Running WITH -DdockerHost specified WITH value**
- Use specified docker host address value and assign it to `sdk.properties`
- If docker address is invalid throw exception with message You suggested above

Notice that customization of dockerHost parameter is now able only in batch mode, I've removed this property from interactive mode assuming that we don't want to confuse people who doesn't use custom docker host addresses.